wp-parsely.php

```diff
@@ -67,6 +62,51 @@ public static function set_version( string $version ): void {
 		self::$version = $version;
 	}
 
+	/**
+	 * Returns configuration which represents how the plugin is configured in site.
+	 *
+	 * @return ParselyConfigs
+	 */
+	public static function get_configs() {
+		if ( ! self::is_active() ) {
+			return null;
+		}
+
+		/**
+		 * Variable.
+		 *
+		 * @var ParselyConfigs
+		 */
+		$configs = new stdClass();
+		$options = self::get_parsely_options() ?: [];
+
+		$configs->is_pinned_version            = has_filter( 'wpvip_parsely_version' );
+		$configs->site_id                      = $options['apikey'] ?? '';
+		$configs->have_api_secret              = '' !== ( $options['api_secret'] ?? '' );
+		$configs->is_javascript_disabled       = (bool) ( $options['disable_javascript'] ?? false );
+		$configs->is_autotracking_disabled     = (bool) ( $options['disable_autotrack'] ?? false );
+		$configs->should_track_logged_in_users = (bool) ( $options['track_authenticated_users'] ?? false );
+
+		$configs->tracked_post_types = array();
+		$post_names                  = get_post_types( array( 'public' => true ) );
+		foreach ( $post_names as $post_name ) {
+			$tracked_post_type       = new stdClass();
+			$tracked_post_type->name = $post_name;
+
+			if ( in_array( $post_name, ( $options['track_post_types'] ?? array() ) ) ) {
+				$tracked_post_type->track_type = 'post';
+			} elseif ( in_array( $post_name, ( $options['track_page_types'] ?? array() ) ) ) {
```

Comment:
I would assign `( $options['track_post_types'] ?? array() )` and `( $options['track_page_types'] ?? array() )` to variables outside to the loop. Improves readability and avoids the needs to create unnecessary arrays on every iteration of the loop.